File: Users/Users.App/Interface/IUserRepo.cs
﻿using Users.Domain;

namespace Users.App.Interface
{
    public interface IUserRepo
    {
        Task AddAsync(User user);

        Task<User?> FindByUserNameAsync(string userName);

        Task<User?> FindByIdAsync(Guid id);

        Task UpdateAsync(User user);

        void Migrate();
    }
}

File: NodeBackApi/src/repos/interfaces/IUserRepo.ts
import type { User } from "../../models/User";

export interface IUserRepo {
  insertAsync(user: User): Promise<void>;
  updateAsync(user: User): Promise<void>;
  findByIdAsync(id: string): Promise<User | null>;
  findByUserNameAsync(userName: string): Promise<User | null>;
  findByEmailAsync(email: string): Promise<User | null>;
  findByExternalIdAsync(externalId: string): Promise<User | null>;
  findByConfirmationTokenAsync(token: string): Promise<User | null>;
  findByResetPasswordTokenAsync(token: string): Promise<User | null>;
}
